src/app/modules/registration/components/water/simple-water-obs/simple-water-obs.component.ts

```diff
@@ -48,16 +73,44 @@ export class SimpleWaterObsComponent {
   }
 
   async save(): Promise<void> {
-    const isEmpty = isObservationModelEmptyForRegistrationTid(
-      this.draft.registration,
-      RegistrationTid.GeneralObservation
-    );
-    if (isEmpty) {
-      this.draft.registration.GeneralObservation = null;
-    }
+    const isEmpty = await this.isSimpleWaterGeneralObsEmpty(this.draft);
+    // if genObs is not empty (for example there are attachments) but self GeneralObservation property
```

Comment:
Dette tror jeg du kan løse på en enklere måte. 
Hva om du endrer isObservationModelEmptyForRegistrationTid() i registration-helper.ts slik at denne også funker for GeneralObs? 
Da tror jeg du kan la denne metoden være som den var.
Om du løser det slik får tror jeg du heller ikke trenger å override isEmpty i GeneralCommentPage.